### PR TITLE
main: Wire up --log-level, add some debug logging

### PR DESCRIPTION
**Pull request by cgwalters**
There doesn't seem to be much usage of logrus here, but I find debug logging very useful.

### CODE REVIEW

**Comment by cgwalters** [comment]
(Only lightly tested locally here, I'm on a ✈️ at the moment, and the Go module fetch step ruins things there)

**Review by mvo5 — APPROVED** [review]
Thank you, this looks good and useful. I especially like that we info pulling image now because that tended to be a bit slow and it looked like bib was hanging.

**Comment by cgwalters** [comment]
> I especially like that we info pulling image now because that tended to be a bit slow and it looked like bib was hanging.

Well that didn't get printed as we were only logging at error level, not info.  I changed it to be an unconditional print for now.

**Comment by cgwalters** [comment]
Looks like testing farm disk space issues again
> processing tar file(write /big-file3: no space left on device): exit status 1

File: bib/cmd/bootc-image-builder/main.go [hunk 1]
```diff
@@ -139,10 +140,13 @@ func makeManifest(c *ManifestConfig, cacheRoot string) (manifest.OSBuildManifest
 	// to using containers storage in all code paths happened.
 	// We might want to change this behaviour in the future to match podman.
 	if !c.Local {
+		fmt.Printf("Pulling image %s (arch=%s)\n", c.Imgref, c.Architecture.String())
```
Comment (mvo5) [inline bib/cmd/bootc-image-builder/main.go:143]: Sorry, I just realized (after looking at the failing tests) this has the same issue as: https://github.com/osbuild/bootc-image-builder/pull/286#discussion_r1530421631 - when we use "--manifest" we expect a pure json output. 

So I guess we need to either use stderr (which is weird as it's not an error) or use logrus and in manifest mode set the level to error but in build mode to info (or something else I have not thought about :)

**Comment by cgwalters** [comment]
(I thought I'd adjusted the `fmt.Printf`, but now finally fixed)

**Comment by cgwalters** [comment]
This one isn't critical but could probably go in, right?

**Review by mvo5 — DISMISSED** [review]
I like it! Thanks for this PR! (I will add a small smoke test for it tomorrow)

**Review by mvo5 — APPROVED** [review]
This :+1: I repushed to fix trivial merge conflicts.

**Pull request merged**